server/src/main/java/com/objectcomputing/checkins/services/private_notes/PrivateNoteController.java

```diff
@@ -0,0 +1,85 @@
+package com.objectcomputing.checkins.services.private_notes;
+
+import com.objectcomputing.checkins.services.role.RoleType;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.Error;
+import io.micronaut.http.annotation.*;
+import io.micronaut.http.hateoas.JsonError;
+import io.micronaut.http.hateoas.Link;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.util.UUID;
+
+@Controller("/services/private-note")
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Produces(MediaType.APPLICATION_JSON)
+@Tag(name = "private-note")
+public class PrivateNoteController {
+
+    private final PrivateNoteServices privateNoteServices;
+
+    public PrivateNoteController(PrivateNoteServices privateNoteServices) {
+        this.privateNoteServices = privateNoteServices;
+    }
+
+    @Error(exception = PrivateNotesBadArgException.class)
+    public HttpResponse<?> handleBadArgs(HttpRequest<?> request, PrivateNotesBadArgException e) {
+        JsonError error = new JsonError(e.getMessage())
+                .link(Link.SELF, Link.of(request.getUri()));
+
+        return HttpResponse.<JsonError>badRequest()
+                .body(error);
+    }
+
+    /**
+     * Create and Save a new check in private note
+     *
+     * @param privateNote
+     * @param request
+     * @return
+     */
+    @Post("/")
+    @Secured({RoleType.Constants.MEMBER_ROLE, RoleType.Constants.PDL_ROLE})
+    public HttpResponse<PrivateNote> createPrivateNote(@Body @Valid PrivateNoteCreateDTO privateNote, HttpRequest<PrivateNoteCreateDTO> request) {
+        PrivateNote newPrivateNote = privateNoteServices.save(new PrivateNote(privateNote.getCheckinid(), privateNote.getCreatedbyid()
+                , privateNote.getDescription()));
+        return HttpResponse.created(newPrivateNote)
+                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getPath(), newPrivateNote.getId()))));
+
```

Comment:
Should this new service be made non-blocking?

Reply:
Yeah, let's convert this one now rather than adding another in this way.